apply new, more generic analysis format (#10)
CERT-Polska/karton-config-extractor: README.md

},
{
    "type": "analysis",
},
```

> For the analysis type, the extractor tries to retrieve config from each memory dump passed in `dumps.zip` resource, but will pick only the best candidate from each family.

Author: I've added some `analysis` task format description to README. Let me know if it's ok, or should I extend it.

},
{
    "type": "analysis",
},
```

While `sample` type is self explanatory, the `analysis` type might be confusing. The `analysis` task is an output from
one of sandboxes: `drakvuf-sandbox`, `cuckoo`, or `joesandbox`. Analysis is a `sample` with additional memory dumps
attached.

The `analysis` type task is expected to be in format:
```
task = Task(
    headers={"type": "analysis"}
    payloads={
        "dumps_metadata": [

>             {"filename": <dump1_filename>, "base_addr": <dump1_baseaddr>},
>             {"filename": <dump2_filename>, "base_addr": <dump2_baseaddr>},
>             {"filename": <dump3_filename>, "base_addr": <dump3_baseaddr>},
>             [...]],

Reviewer: It's `base_address` not `base_addr`
```suggestion
            {"filename": <dump1_filename>, "base_address": <dump1_baseaddr>},
            {"filename": <dump2_filename>, "base_address": <dump2_baseaddr>},
            {"filename": <dump3_filename>, "base_address": <dump3_baseaddr>},
            [...]],
```

In addition, there's no `resources` attribute in `Task()` (not included in suggestion). All keys are resources. It would be nice to see here more real-life example that uses `Resource.from_directory`.

If you need more hints, you can check our internal PR with Cuckoo integration (that code will be probably OSSed soon as well)

Author: Good catch:)

        },
    resources={
        "sample": <sample>
        "dumps.zip": <packed memory dumps in zip format>
    }
)
```

> where `dumps_metadata` contains information about filename and base address for every memory dump in `dumps.zip`.

Reviewer 2: I'd add something like

`baseaddr` is a string with a hex-encoded base address of the dump (leading `0x` is supported)

Author: Done, applied @psrok1 suggestion.

> where `dumps_metadata` contains information about filename and base address for every memory dump in `dumps.zip`.

Reviewer: ```suggestion
where `dumps_metadata` contains information about every memory dump in `dumps.zip`. The following attributes are:
- `filename` which is relative path to the dumps.zip;
- `base_address` which hex-encoded base address for dump (leading `0x` is supported)

You can specify multiple entries for the same file if the same memory dump was found on different base addresses.
```

Reviewer 2: `relative path to the dumps.zip` relative is a key word kere :D. Actually, what if I put `/etc/passwd` there?

Reviewer: Oh :see_no_evil: That should be spotted during https://github.com/CERT-Polska/karton-config-extractor/commit/ab6ad4ef6e773e094ef8f78e183f8e8fbe4dd2ea review.

},
{
    "type": "analysis",
},
```

While `sample` type is self explanatory, the `analysis` type might be confusing. The `analysis` task is an output from
one of sandboxes: `drakvuf-sandbox`, `cuckoo`, or `joesandbox`. Analysis is a `sample` with additional memory dumps
attached.

The `analysis` type task is expected to be in format:
```
task = Task(
    headers={"type": "analysis"}

>     payloads={

Reviewer: ```suggestion
    payload={
```
